src/FEValues/interface_values.jl

```diff
@@ -0,0 +1,305 @@
+            f_value_here = $(f_)(iv, qp, u_a, use_element_a = true)
+            f_value_there = $(f_)(iv, qp, u_b, use_element_a = false)
+            fv = iv.face_values_a
+            result = 0.5 * f_value_here
+            fv = iv.face_values_b
+            result += 0.5 * f_value_there
+            return result
+        end
+    end
+end
+
+for (func,                          f_,                 ) in (
+    (:function_value_jump,          :function_value,    ),
+    (:function_gradient_jump,       :function_gradient, ),
+)
+    @eval begin
+        function $(func)(iv::InterfaceValues, qp::Int, u_a::AbstractVector, u_b::AbstractVector, dof_range_a = eachindex(u_a), dof_range_b = eachindex(u_b))
+            f_value_here = $(f_)(iv, qp, u_a, dof_range_a, use_element_a = true)
+            f_value_there = $(f_)(iv, qp, u_b, dof_range_b, use_element_a = false)
+            return f_value_here - f_value_there 
+        end
+        # TODO: Deprecate this, nobody is using this in practice...
+        function $(func)(iv::InterfaceValues, qp::Int, u_a::AbstractVector{<:Vec}, u_b::AbstractVector{<:Vec})
+            f_value_here = $(f_)(iv, qp, u_a, use_element_a = true)
+            f_value_there = $(f_)(iv, qp, u_b, use_element_a = false)
+            return f_value_here - f_value_there 
+        end
+    end
+end
+
+"""
+    transform_interface_point(iv::InterfaceValues, point::AbstractArray)
+
+Transform point from element A's face reference coordinates to element B's face reference coordinates.
+"""
+function transform_interface_point(iv::InterfaceValues, point::Vec{N, Float64}) where {N}
+    ioi = iv.ioi[]
+    cell = getcells(iv.grid)[iv.cell_a_idx[]]
+    face = iv.face_values_a.current_face[]
+    point = transfer_point_cell_to_face(point, cell, face)
+    N == 3 && (point = (ioi.transformation * Vec(point[1],point[2], 1.0))[1:2])
+    ioi.flipped && (point = reverse(point))
+    N == 3 && return transfer_point_face_to_cell(Vec(point[1],point[2]), getcells(iv.grid)[iv.cell_b_idx[]], iv.face_values_b.current_face[])
+    return transfer_point_face_to_cell(Vec(point[1]), getcells(iv.grid)[iv.cell_b_idx[]], iv.face_values_b.current_face[])
+end
+
+function transform_interface_point(iv::InterfaceValues, point::Vector{Vec{N, Float64}}) where {N}
+    ioi = iv.ioi[]
+    cell = getcells(iv.grid)[iv.cell_a_idx[]]
+    face = iv.face_values_a.current_face[]
+    point2 = transfer_point_cell_to_face.(point, Ref(cell), Ref(face))
+    if N == 3
+        M = ioi.transformation * vcat(reduce(hcat, point2), ones(Float64, 1, length(point)))
+        point2 = [(M[1:2,i]) for i in 1:length(point)]
+    end
+    ioi.flipped && (point2 .= reverse.(point2))
+    N == 3 && return transfer_point_face_to_cell.(point2, Ref(getcells(iv.grid)[iv.cell_b_idx[]]), Ref(iv.face_values_b.current_face[]))
+    return transfer_point_face_to_cell.(point2, getcells(iv.grid)[iv.cell_b_idx[]], iv.face_values_b.current_face[])
+end
```

Comment:
These functions are called from within `reinit!` and they are the only functions which access the grid field of `InterfaceValues`, right? Can't we just pass the two elements into these routines instead of carrying the full grid?